validator_manager/src/list_validators.rs

```diff
@@ -0,0 +1,203 @@
+use clap::{Arg, ArgAction, ArgMatches, Command};
+use eth2::SensitiveUrl;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+use crate::{common::vc_http_client, DumpConfig};
+
+pub const CMD: &str = "list";
+pub const VC_URL_FLAG: &str = "vc-url";
+pub const VC_TOKEN_FLAG: &str = "vc-token";
+
+pub fn cli_app() -> Command {
+    Command::new(CMD)
+        .about("Lists all validators in a validator client using the HTTP API.")
+        .arg(
+            Arg::new(VC_URL_FLAG)
+                .long(VC_URL_FLAG)
+                .value_name("HTTP_ADDRESS")
+                .help(
+                    "A HTTP(S) address of a validator client using the keymanager-API. \
+            If this value is not supplied then a 'dry run' will be conducted where \
+            no changes are made to the validator client.",
+                )
+                .default_value("http://localhost:5062")
+                .requires(VC_TOKEN_FLAG)
+                .action(ArgAction::Set)
+                .display_order(0),
+        )
+        .arg(
+            Arg::new(VC_TOKEN_FLAG)
+                .long(VC_TOKEN_FLAG)
+                .value_name("PATH")
+                .help("The file containing a token required by the validator client.")
+                .action(ArgAction::Set)
+                .display_order(0),
+        )
+}
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub struct ListConfig {
+    pub vc_url: SensitiveUrl,
+    pub vc_token_path: PathBuf,
+}
+
+impl ListConfig {
+    fn from_cli(matches: &ArgMatches) -> Result<Self, String> {
+        Ok(Self {
+            vc_token_path: clap_utils::parse_required(matches, VC_TOKEN_FLAG)?,
+            vc_url: clap_utils::parse_required(matches, VC_URL_FLAG)?,
+        })
+    }
+}
+
+pub async fn cli_run(matches: &ArgMatches, dump_config: DumpConfig) -> Result<(), String> {
+    let config = ListConfig::from_cli(matches)?;
+    if dump_config.should_exit_early(&config)? {
+        Ok(())
+    } else {
+        println!("{}", run(config).await?);
+        Ok(())
+    }
+}
+
+async fn run<'a>(config: ListConfig) -> Result<String, String> {
+    let ListConfig {
+        vc_url,
+        vc_token_path,
+    } = config;
+
+    let (_, validators) = vc_http_client(vc_url.clone(), &vc_token_path).await?;
+
+    let mut result = String::new();
+
+    result.push_str(format!("List of validators ({}):", validators.len()).as_str());
+    result.push('\n');
+    for validator in validators {
+        result.push_str(format!("{}", validator.validating_pubkey).as_str());
+        result.push('\n');
+    }
+
+    Ok(result)
```

Comment:
It's a bit strange that we return a string here, but never do anything with it other than `string.contains(pubkey)`. We could return the `validators` which is a `Vec<SingleKeystoreResponse>` instead, which is cleaner and simpler.

Reply:
This is revised in [c652724](https://github.com/sigp/lighthouse/pull/6261/commits/c652724410021ed638620f6e2cadda3fc9263028), thank you for your help! 

I think I understand what you mean, now that we return Vec<SingleKeystoreResponse> from the `run` function`, which we can obtain the `validating_pubkey` from one of the field in the struct, which is more natural. We can then use that to compare with local public keys. 

Feel free to point out anything that's still weird or incorrect